src/libraries/System.Threading.Tasks.Dataflow/src/Blocks/TransformManyBlock.IAsyncEnumerable.cs

```diff
@@ -86,19 +84,18 @@ public TransformManyBlock(Func<TInput, IAsyncEnumerable<TOutput>> transform, Exe
             // Handle async cancellation requests by declining on the target
             Common.WireCancellationToComplete(
                 dataflowBlockOptions.CancellationToken, Completion, state => ((TargetCore<TInput>)state!).Complete(exception: null, dropPendingMessages: true), _target);
-#if FEATURE_TRACING
+
             DataflowEtwProvider etwLog = DataflowEtwProvider.Log;
             if (etwLog.IsEnabled())
             {
                 etwLog.DataflowBlockCreated(this, dataflowBlockOptions);
             }
-#endif
         }
 
         /// <summary>Processes the message with a user-provided transform function that returns an async enumerable.</summary>
         /// <param name="transformFunction">The transform function to use to process the message.</param>
         /// <param name="messageWithId">The message to be processed.</param>
-        private void ProcessMessage(Func<TInput, IAsyncEnumerable<TOutput>> transformFunction, KeyValuePair<TInput, long> messageWithId)
+        private async void ProcessMessage(Func<TInput, IAsyncEnumerable<TOutput>> transformFunction, KeyValuePair<TInput, long> messageWithId)
```

Comment:
This needs to be awaited as well, or some other mechanism for communicating to the TargetCore that the operations associated with this message have been handled.  Take a look at the existing Task-based async transform to get a sense for what I mean.

Reply:
I have added calls to `_target.Complete` as done in existing Task-based async transform.
Is it enough?